kyaml/kio/filters/merge3.go

```diff
@@ -19,17 +19,18 @@ const (
 	mergeSourceDest       = "dest"
 )
 
+// ResourceMatcher interface is used to match two resources based on IsSameResource implementation
```

Comment:
Nit: We should explain here how this interface is used and why users might want to provide different implementations.

Reply:
@mortent Updated.